vantage6-client/vantage6/client/algorithm_client.py

```
@@ -0,0 +1,296 @@
+                "name": name,
+                "image": self.image,
+                "collaboration_id": self.collaboration_id,
+                "description": description,
+                "organizations": organization_json_list,
+                "database": self.database
+            })
+
+    class VPN(ClientBase.SubClient):
+        """
+        A VPN client for the algorithm container.
+
+        It provides functions to obtain the IP addresses of other containers.
+        """
+        def get_addresses(
+            self, include_children: bool = False, include_parent: bool = False,
+            label: str = None
+        ) -> List[dict]:
+            """
+            Get information about the VPN IP addresses and ports of other
+            algorithm containers involved in the current task. These addresses
+            can be used to send VPN communication to.
+
+            Parameters
+            ----------
+            include_children : bool, optional
+                Include the IP addresses of the children of the current task,
+                by default False.
+            include_parent : bool, optional
+                Include the IP address of the parent of the current task, by
+                default False.
+            label : str, optional
+                The label of the port you are interested in, which is set
+                in the algorithm Dockerfile. If this parameter is set, only
+                the ports with this label will be returned.
+
+            Returns
+            -------
+            List[dict]
+                List of dictionaries containing the IP address and port number,
+                and other information to identify the containers.
+            """
+            results = self.request("vpn/algorithm/addresses", params={
+                "include_children": include_children,
+                "include_parent": include_parent,
+                "label": label
+            })
+
+            return results['addresses']
```

Since you are not passing the output of the request directly, this could raise an exception. 

Check that the key is present or the status code. But I dont think we want to raise an exception here.

---

Added check on the presence of the key, otherwise sending a message back to the algorithm.